### PR TITLE
Ciam 1597 device auth flow

### PR DESCRIPTION
**Pull request by zwitter1**

### CODE REVIEW

File: themes/src/main/resources/theme/base/login/info.ftl [hunk 1]
```diff
@@ -6,19 +6,21 @@
...
+    <#elseif section = "form">
+    <div id="kc-info-message" class="${properties.kcLargeMarginTop}">
         <#if skipLink??>
         <#else>
             <#if pageRedirectUri?has_content>
                 <p><a href="${pageRedirectUri}">${kcSanitize(msg("backToApplication"))?no_esc}</a></p>
             <#elseif actionUri?has_content>
-                <p><a href="${actionUri}">${kcSanitize(msg("proceedWithAction"))?no_esc}</a></p>
+                <input type="button" class="${properties.kcButtonClass!} ${properties.kcButtonPrimaryClass!}" onclick="location.href='${actionUri}'" value="${kcSanitize(msg("proceedWithAction"))}">
+                <a class="pf-c-button pf-m-link" href="#">${kcSanitize(msg("learnMoreDeviceAuth"))?no_esc} <i class="fas fa-external-link-alt"></i></a>
```
Comment (JonSzeto821) [inline themes/src/main/resources/theme/base/login/info.ftl:19]: may be worth tucking these class names into the themes.properties file

File: themes/src/main/resources/theme/base/login/login-oauth2-device-verify-user-code.ftl [hunk 2]
```diff
@@ -2,14 +2,20 @@
...
+        ${msg("oauth2DeviceInvalidUserCodeTitle")}
     <#elseif section = "form">
+    
         <form id="kc-user-verify-device-user-code-form" class="${properties.kcFormClass!}" action="${url.oauth2DeviceVerificationAction}" method="post">
-            <div class="${properties.kcFormGroupClass!}">
-                <div class="${properties.kcLabelWrapperClass!}">
-                    <label for="device-user-code" class="${properties.kcLabelClass!}">${msg("verifyOAuth2DeviceUserCode")}</label>
-                </div>
-
+            <div class="${properties.kcFormGroupClass!} pf-u-mt-lg">
```
Comment (JonSzeto821) [inline themes/src/main/resources/theme/base/login/login-oauth2-device-verify-user-code.ftl:14]: place "pf-u-mt-lg" in theme.properties file

**Comment by JonSzeto821** [comment]
Reviewed the front end related files and made a few minor suggestions - otherwise it looks good to me.

**Comment by stelewis-redhat** [comment]
The backend changes look good here.

**Pull request merged**